src/test/java/org/apache/commons/lang3/builder/DiffBuilderTest.java

```diff
@@ -495,4 +519,16 @@ public void testTriviallyEqualTestEnabled() {
         assertThat(explicitTestAndEqual.build().getNumberOfDiffs(), equalToZero);
     }
 
+    @Test
+    public void testNestedDiffable() {
```

Comment:
This test needs to be different: It compiles just fine AND without warnings without the change to `main`. What am I missing?

Reply:
Let me validate again, but this should not be passing. The generic type T in master (did it get renamed to main? I don't see that in the repo) makes it where the passed diff needs to be the same type as the original builder. Making it ? allows other types of DiffBuilders to be passed. If this is not the behavior in master, I can rewrite the test.

Reply:
After checking out main/master and copy-pasting the test without the code fix, I am getting a compilation error due to the generic types, which is what was asked to be fixed in the original issue. The test cannot compile without the NestedTypeTestClass, so I am unsure why main would be working. Screenshot proof prior to the fix is attached.
![DiffBuilderGenericCompileFailure](https://github.com/apache/commons-lang/assets/16594743/db4f4b6b-a05f-4eab-97bd-cb68eb1b68a0)